arm-network/2016-12-01/swagger/applicationGateway.json

```diff
@@ -0,0 +1,1323 @@
+          "description": "Provisioning state of the public IP resource. Possible values are: 'Updating', 'Deleting', and 'Failed'."
+        }
+      },
+      "description": "Properties of Frontend IP configuration of an application gateway."
+    },
+    "ApplicationGatewayFrontendIPConfiguration": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ApplicationGatewayFrontendIPConfigurationPropertiesFormat"
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of the resource that is unique within a resource group. This name can be used to access the resource."
+        },
+        "etag": {
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
+        }
+      ],
+      "description": "Frontend IP configuration of an application gateway."
+    },
+    "ApplicationGatewayFrontendPortPropertiesFormat": {
+      "properties": {
+        "port": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Frontend port"
+        },
+        "provisioningState": {
+          "type": "string",
+          "description": "Provisioning state of the frontend port resource. Possible values are: 'Updating', 'Deleting', and 'Failed'."
+        }
+      },
+      "description": "Properties of Frontend port of an application gateway."
+    },
+    "ApplicationGatewayFrontendPort": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ApplicationGatewayFrontendPortPropertiesFormat"
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of the resource that is unique within a resource group. This name can be used to access the resource."
+        },
+        "etag": {
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
+        }
+      ],
+      "description": "Frontend port of an application gateway."
+    },
+    "ApplicationGatewayBackendAddress": {
+      "properties": {
+        "fqdn": {
+          "type": "string",
+          "description": "Fully qualified domain name (FQDN)."
+        },
+        "ipAddress": {
+          "type": "string",
+          "description": "IP address"
+        }
+      },
+      "description": "Backend address of an application gateway."
+    },
+    "ApplicationGatewayBackendAddressPoolPropertiesFormat": {
+      "properties": {
+        "backendIPConfigurations": {
+          "type": "array",
+          "items": {
+            "$ref": "./networkInterface.json#/definitions/NetworkInterfaceIPConfiguration"
```

Comment:
which is true. Because there is no networkInterface.json file in the `2016-12-01/swagger` folder. It is present in `2016-09-01/swagger` folder.

The bigger problem here is:
- Can we reference the "NetworkInterfaceIPConfiguration" model from a swagger that is having api-version "2016-09-01"?
  - If yes then wouldn't it break the theory that "api-version" is a contract. One should return models that represent the schema as per that api-version
- If one copies over the "networkInterface.json" file to `2016-12-01` api-version folder then is NRP ready to provision that version of the NetworkInterface API in all the regions in AzureProd ?
  - If NRP is indeed ready with this api-version for networkInterface (i.e. they have already deployed this api-version) then why have they not sent a PR with an updated swagger spec for the new api-version ("2016-12-01").